dev/core#4251 adds support for writing contact image_URL from file en…

JSProffitt opened the pull request:
Overview
----------------------------------------
Adds FK for the contact image file reference and adds/updates the image_URL 

Before
----------------------------------------
image_URL is not created when files are uploaded outside of the contact edit forms.

After
----------------------------------------
Contact Image edits via Afform and other methods now add/update the image_URL

JSProffitt commented:
@colemanw - PR for testing the image_url update

JSProffitt commented:
test this please

eileenmcnaughton commented on CRM/Contact/BAO/Contact.php:
```diff
@@ -1137,6 +1169,24 @@ public static function processImageParams(
+   */
+  public static function hook_civicrm_pre($op, $objectName, $id, &$params): void {
+    // Add or update the contact image private URL in civicrm_contact
+    if (in_array($op, ['create', 'edit']) && \Civi\Api4\Utils\CoreUtil::isContact($objectName)) {
```
Wouldn't this always be a contact cos we are on the contact object?

```suggestion
    if (in_array($op, ['create', 'edit']) ) {
```

colemanw commented:
Fun fact I learned while working on this PR. `self_hook_civicrm_pre` doesn't work in the Contact BAO, because the hook is called for "Individual", "Organization", etc not "Contact" so the `self_` bit of magic fails to match. FYI @totten @eileenmcnaughton

JSProffitt committed:
dev/core#4251 adds support for writing contact image_URL from file entity

also: updated hook to standard pre. Updated image_url label and input type in FormBuilder

JSProffitt committed:
fixed formatting. Updated test field label

JSProffitt committed:
Merge pull request #1 from JSProffitt/contact-image-update

fixed formatting. Updated test field label

JSProffitt committed:
Merge branch 'master' into master

kcristiano commented:
Testing notes:

Test A:   New Install of 6.4.alpha + PR

- New DB Columns display
- No old data to review
- Contact Summary Screen shows:     
<img width="610" alt="image" src="https://github.com/user-attachments/assets/83df1d4a-59af-4ea2-aca4-2cc2cd3e0bf5" />
- [ ] There is No way to upload via contact Summary screen - Should we have Image URL field?
- Profiles - Add Field to profile -- 
<img width="596" alt="image" src="https://github.com/user-attachments/assets/4b379a84-d364-4be0-b222-d4a7da2b086d" />
- [ ]Only shows Image URL  - is this expected?
- using profile allows upload and can see on Contact Summary  
<img width="675" alt="image" src="https://github.com/user-attachments/assets/5cc654c1-7949-4582-bcde-18a6e76190e9" />
- Created Form Builder Two Image options were available - 
<img width="260" alt="image" src="https://github.com/user-attachments/assets/3cc8b067-5b18-4d61-8415-f94d03849d06" />
- These display as 
<img width="646" alt="image" src="https://github.com/user-attachments/assets/885e3d8e-2f64-444a-9614-00a73c51378e" />
- [ ] Can we limit this to just the Image URL
- Using Form in Create Mode works
- [ ] Using Form Builder in update causes console errors and spinner of death

```
TypeError: can't access property "error_message", error is undefined
    Angular 6
        loadData
        k
        $digest
        evalAsync
        completeTask
        d
 Possibly unhandled rejection: {} angular.js:15697:15
    Angular 7
        e
        get
        g
        $digest
        evalAsync
        completeTask
        d
```
​- Directory listing with images:   Uing mix of images uploaded via profiles and Form Builder
<img width="521" alt="image" src="https://github.com/user-attachments/assets/f680c915-34f5-46b7-8ea6-44ef2b71f277" />
- Needed to use Image column to display image - URL Field will only display URL
- Images uploaded via profile will not display as images as shown in screen shot


I started on Test B - Upgrade of existing site, but encountered the above issues plus I had some issues with existing SK Displays with listings.  As this is a test site I need to dig in and be sure this was this PRs issue and not my enviornment.
